lib/calendar_recurrence/rrule.ex

```diff
@@ -372,4 +372,79 @@ defmodule CalendarRecurrence.RRULE do
         %{current | day: next_day}
     end
   end
+
+  defimpl String.Chars do
+    @doc """
+    Converts `%RRULE{}` into a rrule string.
+
+    ## Examples
+
+        iex> to_string(%RRULE{freq: :daily, count: 2})
+        "FREQ=DAILY;COUNT=2"
+
+    """
+    @spec to_string(CalendarRecurrence.RRULE.t()) :: binary()
+    def to_string(rrule) do
+      rrule = Map.from_struct(rrule)
+
+      [
+        :freq,
+        :interval,
+        :until,
+        :count,
+        :bysecond,
+        :byminute,
+        :byhour,
+        :byday,
+        :bymonthday,
+        :byyearday,
+        :bymonth
+      ]
+      |> Enum.reduce([], fn key, acc -> [add_part(key, rrule[key]) | acc] end)
+      |> Enum.reverse()
+      |> Enum.reject(&(&1 == []))
+      |> Enum.intersperse(";")
+      |> IO.iodata_to_binary()
+    end
+
+    @weekdays %{
+      1 => "MO",
+      2 => "TU",
+      3 => "WE",
+      4 => "TH",
+      5 => "FR",
+      6 => "SA",
+      7 => "SU"
+    }
+
+    defp add_part(_key, nil), do: []
+    defp add_part(_key, []), do: []
+    defp add_part(:interval, 1), do: []
+
+    defp add_part(:until = key, value) do
```

Comment:
I would guard this because the until field can be:

```
@type date() :: Date.t() | NaiveDateTime.t() | DateTime.t() | CalendarRecurrence.T.t()
```

Can we support the other Date/Time formats according to the spec?

```suggestion
    defp add_part(:until = key, %DateTime{} = value) do
```